### ISSUE TITLE
Standardize naming in loaders

### ISSUE THREAD

**Issue by 23pointsNorth** [Deci-AI/super-gradients#1425]
### 🚀 Feature Request

There has been a few mismatches between the naming conventions and input params. It would be great to standardize those, as to make code re-use much easier. To name a few:
- [ ] In COCO segmentation/deteciton there is a difference between the loader expecting `data_dir` and `root_dir`. 
- [ ] The classes capitalization is following different standards as well `COCODetectionDataset` vs `CoCoSegmentationDataSet`.
- [ ] There is `coco_segmentation_dataset_params`, but no `coco_detection_dataset_params`, but `coco_detection_{network structure}_dataset_params`. Change the segmentation to follow the same standard? 
- [ ] Similarly `dataloaders.coco_segmentation_train` vs `dataloaders.coco2017_train_yolo_nas`.
- [ ] define a `__str__` representation for the loaders, to avoid `<torch.utils.data.dataloader.DataLoader object at 0x7fadc1d16620>` 
- [ ] segmentation has `train_loader.dataset.dataset_classes_inclusion_tuples_list`, but detection doesn't. A common set of params?

edit: (a few more)

- [ ] DetectionMetrics, PoseEstimationMetrics, and `PreprocessSegmentationMetricsArgs`
- [ ] In tutorials/colabs - train_loader, train_dataloader, train_data, val_loader, valid_loader.
- [ ] `coco2017_rescoring_train` is actually using `coco2017_pose_rescoring_*`
- [ ] Fix `Fixme` in [documentation](https://docs.deci.ai/super-gradients/V3_1/docstring/training/losses.html#V3_1.src.super_gradients.training.losses.iou_loss.GeneralizedIoULoss)
- [ ] In link above `PoseEstimationMetrics` comment/code section error.
- [ ] the `metrics_to_watch` is `"target_IoU"` for `*_metrics_list` that includes `BinaryIoU()`, but the metric is just `"IoU"` if it's a multiclass case with the metrics list including `IoU()`.

**Comment by BloodAxe** [comment]
Hey. Thanks for the report. Indeed, the naming scheme is not consistent across the config files & models.
And in fact we already started looking into standardizing it. At this point it is hard to give an accurate estimate when exactly this will be introduced. 

But you may keep an eye on https://github.com/Deci-AI/super-gradients/pull/1421 ;)
